### PR TITLE
Make Retry annotation declarable on spec classes

### PR DESCRIPTION
**Pull request by marcphilipp**
Implementation of option (1) for #874.

### CODE REVIEW

File: spock-specs/src/test/groovy/org/spockframework/smoke/extension/RetryFeatureExtensionSpec.groovy [hunk 1]
```diff
@@ -452,4 +452,79 @@ class Foo extends Specification {
...
+  }
+}
+    """)
+
+    then:
+    result.runCount == 3
+    result.failureCount == 4 + 0 + 2
+  }
+
+  def "@Retry declared on a spec class is inherited"() {
```
Comment (leonard84) [inline spock-specs/src/test/groovy/org/spockframework/smoke/extension/RetryFeatureExtensionSpec.groovy]: You are missing the inverse case plain superSpec annotated childSpec

File: spock-core/src/main/java/spock/lang/Retry.java [hunk 2]
```diff
@@ -27,12 +27,23 @@
...
  *
+ * <p>Retries can be applied to feature methods and spec classes. Applying it to
+ * a spec class has the same effect as applying it to each feature method that
+ * isn't already annotated with {@code Retry}.
+ *
+ * <p>{@code Retry} annotations are inherited, i.e. if declared on a superclass
+ * of a spec class, it will be applied to all feature methods of both classes
+ * unless the spec class declared its own annotation. If so, it is applied to
+ * all feature methods declared in that class, but not to any feature methods
+ * declared in superclasses.
```
Comment (leonard84) [inline spock-core/src/main/java/spock/lang/Retry.java]: Add something to explain that `@Retry` also does not apply to inherited features

File: docs/extensions.adoc [hunk 3]
```diff
@@ -254,6 +254,23 @@ class FlakyIntegrationSpec extends Specification {
 }
 ----
 
+Retries can also be applied to spec classes which has the same effect as applying it to each feature method that isn't
```
Comment (leonard84) [inline docs/extensions.adoc]: Maybe add more of the javadoc description regarding the inheritance

File: spock-core/src/main/java/org/spockframework/runtime/extension/builtin/RetryExtension.java [hunk 4]
```diff
@@ -18,13 +18,24 @@
 
 import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
 import org.spockframework.runtime.model.FeatureInfo;
+import org.spockframework.runtime.model.SpecInfo;
 import spock.lang.Retry;
+import spock.lang.Timeout;
```
[the remark this answers is not in the document]
Comment (marcphilipp) [reply spock-core/src/main/java/org/spockframework/runtime/extension/builtin/RetryExtension.java]: oops. 🙈

**Comment by marcphilipp** [comment]
I addressed the review comments and force-pushed an amended commit.

@leonard84 PTAL! 🙂

**Comment by marcphilipp** [comment]
Closing in favor of #879.

**Pull request closed without merging**